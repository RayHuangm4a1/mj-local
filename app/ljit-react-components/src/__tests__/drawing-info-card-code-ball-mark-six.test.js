import React from 'react';
import { shallow, mount, } from 'enzyme';
import CodeBall from '../components/code-ball';
const CodeBallCircle = CodeBall.Circle;
const MarkSixColorMap = getMarkSixColorMap();
const { SizeEnum, FontSizeEnum, } = CodeBallCircle;

function getMarkSixColorMap() {
	const red = [
		'01', '02', '07', '08', '12', '13', '18', '19', '23', '24', '29', '30', '34', '35', '40', '45', '46',
	];
	const blue = [
		'03', '04', '09', '10', '14', '15', '20', '25', '26', '31', '36', '37', '41', '42', '47', '48',
	];
	const green = [
		'05', '06', '11', '16', '17', '21', '22', '27', '28', '32', '33', '38', '39', '43', '44', '49',
	];
	const ColorMap = {};

	red.forEach(number => {
		ColorMap[number] = '#ff594d';
	});
	blue.forEach(number => {
		ColorMap[number] = '#00a4fc';
	});
	green.forEach(number => {
		ColorMap[number] = '#00d2b7';
	});

	return ColorMap;
}


describe('DrawingInfoCardCodeBallMarkSix', () => {
	let DrawingInfoCardCodeBallMarkSix;

	beforeEach(() => {
		DrawingInfoCardCodeBallMarkSix = require('../components/drawing-info-card/elements/drawing-info-card-code-ball-mark-six').default;
	});

	afterEach(() => {
		DrawingInfoCardCodeBallMarkSix = undefined;
	});

	it('should handle defaultProps.', () => {
		const {
			size,
			fontSize,
		} = DrawingInfoCardCodeBallMarkSix.defaultProps;

		expect(size).toEqual(SizeEnum.MIDDLE);
		expect(fontSize).toEqual(FontSizeEnum.MIDDLE);
	});

	it('should be selectable by class drawing-info-card-code-ball-mark-six', () => {
		const wrapper = shallow(<DrawingInfoCardCodeBallMarkSix
			splitOpencodes={['01','02','03','04','05',]}
		/>);

		expect(wrapper.hasClass('drawing-info-card-code-ball-mark-six')).toEqual(true);
	});

	it('should renders correctly', () => {
		const props = {
			splitOpencodes: ['01','02','03','04','05',],
		};

		const wrapper = shallow(<DrawingInfoCardCodeBallMarkSix {...props} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const props = {
			splitOpencodes: ['01','02','03','04','05',],
		};

		const wrapper = mount(<DrawingInfoCardCodeBallMarkSix {...props} />);

		expect(wrapper.props().opencode).toBe(props.opencode);
		expect(wrapper.props().splitOpencode).toBe(props.splitOpencode);
	});

	it('should render corresponsed ball backgroundColor', () => {
		const props = {
			splitOpencodes: ['01','02','03','04','05',],
		};

		const wrapper = mount(<DrawingInfoCardCodeBallMarkSix {...props} />);
		const wrapperCodeBallCircle = wrapper.find(CodeBallCircle);

		wrapperCodeBallCircle.forEach(codeBall => {
			const { text, backgroundColor, } = codeBall.props().text;

			expect(backgroundColor).toEqual(MarkSixColorMap[text]);
		});
	});
});
