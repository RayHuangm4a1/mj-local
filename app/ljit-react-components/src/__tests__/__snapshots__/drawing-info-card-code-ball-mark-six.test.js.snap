// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawingInfoCardCodeBallMarkSix should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CodeBallMarkSix
    fontSize="middle"
    size="middle"
    splitOpencodes={
      Array [
        "01",
        "02",
        "03",
        "04",
        "05",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Circle
            className="code-ball-mark-six--red"
            fontSize="middle"
            size="middle"
            text="01"
            type="primary"
          />,
          <Circle
            className="code-ball-mark-six--red"
            fontSize="middle"
            size="middle"
            text="02"
            type="primary"
          />,
          <Circle
            className="code-ball-mark-six--blue"
            fontSize="middle"
            size="middle"
            text="03"
            type="primary"
          />,
          <Circle
            className="code-ball-mark-six--blue"
            fontSize="middle"
            size="middle"
            text="04"
            type="primary"
          />,
        ],
        <div>
          +
        </div>,
        <Circle
          className="code-ball-mark-six--green"
          fontSize="middle"
          size="middle"
          text="05"
          type="primary"
        />,
      ],
      "className": "drawing-info-card-code-ball-mark-six",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "01",
        "nodeType": "function",
        "props": Object {
          "className": "code-ball-mark-six--red",
          "fontSize": "middle",
          "size": "middle",
          "text": "01",
          "type": "primary",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "02",
        "nodeType": "function",
        "props": Object {
          "className": "code-ball-mark-six--red",
          "fontSize": "middle",
          "size": "middle",
          "text": "02",
          "type": "primary",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "03",
        "nodeType": "function",
        "props": Object {
          "className": "code-ball-mark-six--blue",
          "fontSize": "middle",
          "size": "middle",
          "text": "03",
          "type": "primary",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "04",
        "nodeType": "function",
        "props": Object {
          "className": "code-ball-mark-six--blue",
          "fontSize": "middle",
          "size": "middle",
          "text": "04",
          "type": "primary",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
        },
        "ref": null,
        "rendered": "+",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "code-ball-mark-six--green",
          "fontSize": "middle",
          "size": "middle",
          "text": "05",
          "type": "primary",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Circle
              className="code-ball-mark-six--red"
              fontSize="middle"
              size="middle"
              text="01"
              type="primary"
            />,
            <Circle
              className="code-ball-mark-six--red"
              fontSize="middle"
              size="middle"
              text="02"
              type="primary"
            />,
            <Circle
              className="code-ball-mark-six--blue"
              fontSize="middle"
              size="middle"
              text="03"
              type="primary"
            />,
            <Circle
              className="code-ball-mark-six--blue"
              fontSize="middle"
              size="middle"
              text="04"
              type="primary"
            />,
          ],
          <div>
            +
          </div>,
          <Circle
            className="code-ball-mark-six--green"
            fontSize="middle"
            size="middle"
            text="05"
            type="primary"
          />,
        ],
        "className": "drawing-info-card-code-ball-mark-six",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "01",
          "nodeType": "function",
          "props": Object {
            "className": "code-ball-mark-six--red",
            "fontSize": "middle",
            "size": "middle",
            "text": "01",
            "type": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "02",
          "nodeType": "function",
          "props": Object {
            "className": "code-ball-mark-six--red",
            "fontSize": "middle",
            "size": "middle",
            "text": "02",
            "type": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "03",
          "nodeType": "function",
          "props": Object {
            "className": "code-ball-mark-six--blue",
            "fontSize": "middle",
            "size": "middle",
            "text": "03",
            "type": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "04",
          "nodeType": "function",
          "props": Object {
            "className": "code-ball-mark-six--blue",
            "fontSize": "middle",
            "size": "middle",
            "text": "04",
            "type": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
          },
          "ref": null,
          "rendered": "+",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "code-ball-mark-six--green",
            "fontSize": "middle",
            "size": "middle",
            "text": "05",
            "type": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
